### PR TITLE
Adds THE SILENCER

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

The silencer is a mime antagonist with the goal to remove the tongues of half of the living station.

## Why It's Good For The Game

I thought the game needed another fastball

## Changelog
:cl:
add: Mimes can now roll for THE SILENCER
/:cl:

### CODE REVIEW

**Comment by epicgamer545** [comment]
Wow, this was out of the blue.

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -692,3 +692,33 @@
...
 	for(var/department_type in department_types)
 		create_separatist_nation(department_type, announcement = FALSE, dangerous = FALSE, message_admins = FALSE)
+
+/datum/dynamic_ruleset/roundstart/silencer
+	name = "Silencer"
+	antag_flag = ROLE_SILENCER
+	antag_datum = /datum/antagonist/silencer
+	flags = LONE_RULESET
+	minimum_required_age = 7
+	/// it's every role but mime. i'm not typing that out
```
Comment (optimumtact) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: there should be a specific empty define like "DEFINED_PROGRAMATICALLY"

that flags to the developer this list is built at a special runtime.

Comment (optimumtact) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: tbh I'm almost on teh fence about it and would just prefer it written out.

or add the opposite of restricted roles (allowed_roles).

Comment (tralezab) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: fair enough, i can implement that

Comment (Mothblocks) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: I think you should set it on `/New()`, to `typesof(/datum/job) - /datum/job/mime`

Comment (Mothblocks) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: Or whatever it is

File: code/modules/antagonists/silencer/silencer.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	mask = /obj/item/clothing/mask/gas/mime
+	l_hand = /obj/item/scalpel
+	r_hand = /obj/item/circular_saw
+
+/datum/antagonist/silencer/on_gain()
+	var/datum/objective/remove_tongues/the_silencers_task = new /datum/objective/remove_tongues("Ensure half of the surviving crew are tongueless.")
+	the_silencers_task.completed = TRUE //YES!
+	the_silencers_task.owner = owner
+	objectives += the_silencers_task
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/antagonists/silencer/silencer.dm:26]:
```suggestion
	return ..()
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
@tgstation/head-admins New antagonist

**Comment by Mothblocks** [comment]
If you want the beret you have to call `update_greyscale` on it, check how assistant jumpsuits do it

**Comment by RaveRadbury** [comment]
This would be better as an option traitor mime-only thing rather than an individual antag to roll for, you're going to have people who wouldn't otherwise play mime rolling it because they have an extra chance to be antag.

That feels bad.

**Comment by EuclidianGeometer** [comment]
i feel like this would be a fun traitor gimmick once and not something i'd want to see on a regular basis, even at it's fairly low weight

**Comment by EuclidianGeometer** [comment]
it's just so specific.

**Comment by lizardqueenlexi** [comment]
I'm a little confused as to how you're even _meant_ to take half the tongues on the station as an otherwise completely normal mime.

**Comment by Mothblocks** [comment]
@RaveRadbury Antags are picked first, I would hope that is not the case

**Comment by itseasytosee** [comment]
> I'm a little confused as to how you're even _meant_ to take half the tongues on the station as an otherwise completely normal mime.

I think I witnessed some discussion somewhere about being able to but people's tonuges out in a similar matter to slitting throats.

**Comment by tralezab** [comment]
> I'm a little confused as to how you're even _meant_ to take half the tongues on the station as an otherwise completely normal mime.

I'll unlock some evil mime abilities. We should just have a way to remove people's tongues, another contributor said they were gonna add it and i'm like alright okay

**Comment by Mothblocks** [comment]
Sharpness + mouth, add toggle for silencer to be able to skip the mouth intent part (mostly as a tutorial for the mechanic)

**Comment by Mothblocks** [comment]
Could be its own PR if u wanted

**Comment by the-orange-cow** [comment]
> Could be its own PR if u wanted

An idea:  instead of an antagonist slot, what about a mime only traitor item, like a dagger or razor, that becomes more powerful as you cut out tongues?  Like the assistant toolbox with tc.

**Comment by RaveRadbury** [comment]
> @RaveRadbury Antags are picked first, I would hope that is not the case

You're right, it still doesn't sit well with me, idk. That would mean that a person rolling with the silencer on but mime low would beat out someone rolling mime high with silencer off. Considering you only get one mime on the station it feels bad.

**Comment by Mothblocks** [comment]
valid

**Comment by necromanceranne** [comment]
Is there anything wrong with a mime in maint that came out wrong? Some uncertainty about which mime is good or bad isn't a bad conflict (especially since they can't talk)

**Comment by timothymtorres** [comment]
Why isn't this a traitor objective that is specifically for mimes?  Would make more sense than having this as a separate antag roll.

**Comment by Shadowflame909** [comment]
the idea feels pretty hollow at the moment.. excited to see this draft come into its own though.

What if the silencer greyscaled/grey crayon dusted the people who he stole tongues from, with some sort of additional effect to create an impact on the round. Like slowing them down or occasionally trapping them in invisible walls on all sides.

I'd also think it'd be neat to have the clown cheer people up and have the ability to remove whatever additional effect + greyness the silencer does to people. It'd make clowns be viewed more fondly. Like how devils were to lawyers, and how slaughter demons are for janitors currently.

Needs soul!

**Comment by Troxonater** [comment]
TBH I would just make this a mime specific thing for prog traitor mime to do instead of being its own antag.

**Comment by itseasytosee** [comment]
You guys have no VISON

**Comment by celulamp** [comment]
Having more antags and variety is way better than putting everything under the traitor umbrella

traitor umbrella is boring! Having more unique antags and variety rather than
traitors every round is good. 

But currently yeah its kinda mid until u give him powers to help
give him unique ones and not just the ones u get from traitor book
(If ppl want to add new antags they could prob just edit old unaccepted ones and fix them up unless the concepts suck)

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm (hunk 1 above)
Comment (Fikou) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:702]: i did that with malf ai and it was horribly broken so idk if thatd work

**Pull request closed without merging**